### ISSUE TITLE
✨ New `ZeroAsShortSerializer` type

### ISSUE THREAD

**Issue by LVMVRQUXL** [kotools/types#694]
## 📝 Description

Similarly to #690, in the `types-kotlinx-serialization` subproject, we want to introduce the `ZeroAsShortSerializer` **experimental** type for serializing the `Zero` type as `Short`.

```kotlin
class ZeroAsShortSerializer : KSerializer<Zero> {
    override val descriptor: SerializerDescriptor

    override fun equals(other: Any?): Boolean
    override fun hashCode(): Int
    override fun serialize(encoder: Encoder, value: Zero)
    override fun deserialize(decoder: Decoder): Zero
    override fun toString(): String
}
```

This serializer should be available for all Kotlin platforms.

## 🔗 Dependencies

This issue is blocked by the following items:

- [x] #716
- [x] #763 
- [x] #784

## ✅ Checklist

- [ ] ✨ Add the type and its public constructor with tests, documentation and samples.
- [ ] ✨ Override the `toString` function with tests, documentation and samples.
- [ ] ✨ Override structural equality operations (`equals` and `hashCode` functions) with tests, documentation and samples.
- [ ] ✨ Make the type implementing the `KSerializer<Zero>` type with tests, documentation and samples.
- [ ] 📝 Update the unreleased changelog for this issue.
- [ ] 📝 After closing this issue, attach tracking issues to a milestone.

**Comment by LVMVRQUXL** [comment]
Instead of creating a new type for each serializations of the `Zero` type, it might be interesting to provide a single `ZeroSerializer` type with factory functions providing the right variant.

```kotlin
interface ZeroSerializer : KSerializer<Zero> {
    companion object {
        fun asByte(): ZeroSerializer = TODO()
        fun asShort(): ZeroSerializer = TODO()
        fun asInt(): ZeroSerializer = TODO()
        fun asLong(): ZeroSerializer = TODO()
        fun asFloat(): ZeroSerializer = TODO()
        fun asDouble(): ZeroSerializer = TODO()
    }
}
```

This idea is also available for #695, #696, #697 and #698.

**Comment by LVMVRQUXL** [comment]
Another alternative to [this comment](https://github.com/kotools/types/issues/694#issuecomment-2681559490) is to provide these factory functions as extension functions directly on the `Zero.Companion` type.

```kotlin
fun Zero.Companion.byteSerializer(): KSerializer<Zero> = TODO()
fun Zero.Companion.shortSerializer(): KSerializer<Zero> = TODO()
fun Zero.Companion.intSerializer(): KSerializer<Zero> = TODO()
fun Zero.Companion.longSerializer(): KSerializer<Zero> = TODO()
fun Zero.Companion.floatSerializer(): KSerializer<Zero> = TODO()
fun Zero.Companion.doubleSerializer(): KSerializer<Zero> = TODO()
```

The main advantage of this solution is that we don't have to provide an additional type, alongside the `KSerializer` interface already provided by the Kotlin Serialization library.

Also, being similar to the `serializer()` function generated by the Kotlin Serialization Gradle plugin when marking a type with the `Serializable` annotation, this could simplify the usage of the `types-kotlinx-serialization` Gradle subproject.

```kotlin
val serializer: KSerializer<Zero> = Zero.byteSerializer()
val zero = Zero()
val encoded: String = Json.encodeToString(serializer, zero)
println(encoded) // 0
val decoded: Zero = Json.decodeFromString(serializer, encoded)
println(zero == decoded) // true
```

**Comment by LVMVRQUXL** [comment]
Superseded by #793.
